Plugin/Magento/Inventory/Model/SourceItem/Command/SourceItemSavePlugin.php

```diff
@@ -16,44 +16,110 @@
  */
 namespace Bolt\Boltpay\Plugin\Magento\Inventory\Model\SourceItem\Command;
 
+use Bolt\Boltpay\Helper\FeatureSwitch\Decider;
 use Bolt\Boltpay\Model\CatalogIngestion\ProductEventProcessor;
+use Magento\Catalog\Model\ProductFactory;
+use Magento\Catalog\Model\ResourceModel\Product\Website\Link as ProductWebsiteLink;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\Module\Manager as ModuleManager;
 use Magento\InventoryApi\Api\SourceItemsSaveInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\InventoryIndexer\Plugin\InventoryApi\ReindexAfterSourceItemsSavePlugin;
 
 /**
  * Catalog ingestion product event processor after source items update
  */
 class SourceItemSavePlugin
 {
+    /**
+     * @var ObjectManager
+     */
+    private $objectManager;
+
+    /**
+     * @var ModuleManager
+     */
+    private $moduleManager;
+
     /**
      * @var ProductEventProcessor
      */
     private $productEventProcessor;
 
+    /**
+     * @var ProductFactory
+     */
+    private $productFactory;
+
+    /**
+     * @var ProductWebsiteLink
+     */
+    private $productWebsiteLink;
+
+    /**
+     * @var Decider
+     */
+    private $featureSwitches;
+
+    /**
+     * @var ReindexAfterSourceItemsSavePlugin
+     */
+    private $reindexAfterSourceItemsSavePlugin;
+
     /**
      * @param ProductEventProcessor $productEventProcessor
+     * @param ProductFactory $productFactory
+     * @param ProductWebsiteLink $productWebsiteLink
+     * @param Decider $featureSwitches
+     * @param ModuleManager $moduleManager
      */
-    public function __construct(ProductEventProcessor $productEventProcessor)
-    {
+    public function __construct(
+        ProductEventProcessor $productEventProcessor,
+        ProductFactory $productFactory,
+        ProductWebsiteLink $productWebsiteLink,
+        Decider $featureSwitches,
+        ModuleManager $moduleManager
+    ) {
+        $this->objectManager = ObjectManager::getInstance();
+        $this->moduleManager = $moduleManager;
         $this->productEventProcessor = $productEventProcessor;
+        $this->productFactory = $productFactory;
+        $this->productWebsiteLink = $productWebsiteLink;
+        $this->featureSwitches = $featureSwitches;
+        if ($this->moduleManager->isEnabled('Magento_InventoryIndexer')) {
+            $this->reindexAfterSourceItemsSavePlugin = $this->objectManager
```

Comment:
Any specific reason why we want to call the plugin directly?
I think it's maybe cleaner if we copy its code to our helpers.

Reply:
we need correct values of product status before and after update, default indexer plugin can't be called in around plugin automatically, but reindex is required to have correct values of product stock status.

Reply:
updated to before/after